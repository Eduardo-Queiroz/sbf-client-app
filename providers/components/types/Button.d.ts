interface ButtonPropsInterface {
  onPress: () => void;
  text: string;
  disabled?: boolean;
}

declare type ButtonProps = ButtonPropsInterface & TrackeablePropsInterface;
