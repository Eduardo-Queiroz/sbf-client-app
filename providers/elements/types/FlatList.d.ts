declare type FlatListProps = import('react-native').FlatListProps<any>;
