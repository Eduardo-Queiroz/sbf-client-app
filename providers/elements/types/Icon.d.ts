interface SizeProps {
  size?: number;
  width?: number;
  height?: number;
}

interface IconPropsInterface extends SizeProps {
  height?: number;
  width?: number;
  size?: number;
  color?: import('@sbf-types/general').Color;
  icon: React.FC<any> | import('@sbf-providers/icons').IconsType;
}

declare type IconProps = IconPropsInterface;
