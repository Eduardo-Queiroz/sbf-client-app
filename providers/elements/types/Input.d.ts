declare type InputProps = import('react-native').TextInputBase &
  TrackeablePropsInterface;
