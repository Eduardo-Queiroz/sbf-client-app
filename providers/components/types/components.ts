export enum ComponentsType {
  LOADING = 'LOADING',
  BUTTON = 'BUTTON',
  FOOTER = 'FOOTER',
}
