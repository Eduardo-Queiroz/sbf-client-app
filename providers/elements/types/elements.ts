export enum ElementsType {
  FLAT_LIST = 'FLAT_LIST',
  FORM = 'FORM',
  ICON = 'ICON',
  LAYOUT_ITEM = 'LAYOUT_ITEM',
  GRID = 'GRID',
  GRID_ITEM = 'GRID_ITEM',
  SAFE_AREA_LAYOUT_ITEM = 'SAFE_AREA_LAYOUT_ITEM',
  SAFE_AREA_GRID_ITEM = 'SAFE_AREA_GRID_ITEM',
  SCROLL_LAYOUT_ITEM = 'SCROLL_LAYOUT_ITEM',
  TOUCHABLE_CONTAINER = 'TOUCHABLE_CONTAINER',
  LOADING = 'LOADING',
  POSITION_ELEMENT = 'POSITION_ELEMENT',
  TOUCHABLE = 'TOUCHABLE',
  TYPOGRAPHY = 'TYPOGRAPHY',
  INPUT = 'INPUT',
  TRACKEABLE_HOC = 'TRACKEABLE_HOC',
  PICKER = 'PICKER',
  MODAL = 'MODAL',
  RATING = 'RATING',
}
