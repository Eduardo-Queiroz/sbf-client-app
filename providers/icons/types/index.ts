export enum IconsType {
  ICON_MINUS = 'ICON_MINUS',
  ICON_PLUS = 'ICON_PLUS',
  ICON_CLOSE = 'ICON_CLOSE',
  ICON_CART = 'ICON_CART',
  ICON_NO_CART = 'ICON_NO_CART',
  ICON_SEARCH = 'ICON_SEARCH',
  ICON_STAR = 'ICON_STAR',
  ICON_SAD_WARNING = 'ICON_SAD_WARNING',
  ICON_HAPPY_WARNING = 'ICON_HAPPY_WARNING',
}
