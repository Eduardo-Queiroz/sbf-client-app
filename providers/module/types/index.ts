export enum ContainerType {
  // BANNER
  BANNER_SINGLE = 'BANNER_SINGLE',
  // CATALOG
  CATALOG_SPOTLIGHT = 'CATALOG_SPOTLIGHT',
  CATALOG_SEARCH = 'CATALOG_SEARCH',
  CATALOG_SUGGESTION = 'CATALOG_SUGGESTION',
  // CART
  CART_BUTTON = 'CART_BUTTON',
  CART_QUANTITY_CONTROL = 'CART_QUANTITY_CONTROL',
}
