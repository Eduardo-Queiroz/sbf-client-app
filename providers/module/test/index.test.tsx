/* eslint-disable react/display-name */
import React from 'react';
import {
  ModuleProvider,
  useContainer,
  useRoutes,
  context,
  ModuleType,
} from './../src/index';
import {ContainerType} from './../types/index';
import {render} from '@testing-library/react-native';
import {renderHook} from '@testing-library/react-hooks';

const valueWrapper = {
  routes: [
    {
      name: true,
    },
  ],
  containers: {
    [ContainerType.BANNER_SINGLE]: {Component: true},
  },
};

const makeWrapper = ({children}: {children?: any}) => (
  <context.Provider value={valueWrapper}>{children}</context.Provider>
);

describe('it should test module provider', () => {
  test('it should verify provider snapshot', () => {
    const {toJSON} = render(
      <ModuleProvider
        modules={[
          {
            name: ContainerType.BANNER_SINGLE,
            type: ModuleType.ROUTE,
            Component: true,
          },
          {
            name: ContainerType.BANNER_SINGLE,
            type: ModuleType.CONTAINER,
            Component: true,
          },
        ]}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('it should verify empty provider context', () => {
    try {
      const {result} = renderHook(() => useRoutes());
      expect(result).toBe(false);
    } catch (e) {
      expect(e.message).toBe('Sem rotas inicialiadas no contexto');
    }
  });

  test('it should verify specific value from provider context', async () => {
    try {
      const {result} = renderHook(() =>
        useContainer(ContainerType.BANNER_SINGLE),
      );
      expect(result).toBe(false);
    } catch (e) {
      expect(e.message).toBe(`Sem containers inicialiados no contexto`);
    }
  });

  test('it should verify specific value from provider route context', async () => {
    const {result} = renderHook(() => useRoutes(), {
      wrapper: makeWrapper,
    });
    expect(result.current).toMatchObject([{name: true}]);
  });

  test('it should verify specific value from provider container context', async () => {
    const {result} = renderHook(
      () => useContainer(ContainerType.BANNER_SINGLE),
      {
        wrapper: makeWrapper,
      },
    );
    expect(result.current).toBe(true);
  });
});
