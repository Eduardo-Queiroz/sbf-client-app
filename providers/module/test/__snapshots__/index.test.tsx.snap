// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should test module provider it should verify provider snapshot 1`] = `null`;
